src/mongocxx/collection.cpp

```diff
@@ -319,17 +323,20 @@ stdx::optional<result::insert_one> collection::insert_one(view_or_value document
 stdx::optional<result::replace_one> collection::replace_one(view_or_value filter,
                                                             view_or_value replacement,
                                                             const options::update& options) {
-    class bulk_write bulk_op(false);
-    model::replace_one replace_op(filter, replacement);
+    options::bulk_write bulk_opts;
+    bulk_opts.ordered(false);
+
+    if (options.bypass_document_validation())
```

Comment:
is there a way to DRY this?  What if we allowed construction of a bulk_write options object using an existing update/insert/delete options object?